src/edu/stanford/nlp/math/SloppyMath.java

```diff
@@ -294,13 +294,8 @@ public static double logAdd(double lx, double ly) {
       max = ly;
       negDiff = lx - ly;
     }
-    if (max == Double.NEGATIVE_INFINITY) {
-      return max;
-    } else if (negDiff < -LOGTOLERANCE) {
-      return max;
-    } else {
-      return max + Math.log(1.0 + Math.exp(negDiff));
-    }
+    return (max == Double.NEGATIVE_INFINITY || negDiff < -LOGTOLERANCE) ? max : //
```

Comment:
Does this have anything to do with efficiency, or just more compact because of ?: operator?  why the // at the end?

Reply:
The // is to prevent code formatters from making this a single line. This is my usual code style with ternary statements to make them more readable.

Ternary statements, while logically being the same and frowned upon by some purists, tend to be faster in my experiments with Java. This may be because of two factors:
1. Smaller byte code size (58 vs. 62)
2. Single return, which may make inlining easier, too.

Reply:
Interesting.  Thanks for the explanation